setuptools_git_versioning.py

```diff
@@ -107,29 +151,14 @@ def parse_config(dist, _, value):  # type: (Distribution, Any, Any) -> None
     if not isinstance(value, collections_abc.Mapping):
         raise DistutilsSetupError("Config in the wrong format")
 
-    template = value.get("template", DEFAULT_TEMPLATE)
-    dev_template = value.get("dev_template", DEFAULT_DEV_TEMPLATE)
-    dirty_template = value.get("dirty_template", DEFAULT_DIRTY_TEMPLATE)
-    starting_version = value.get("starting_version", DEFAULT_STARTING_VERSION)
-    version_callback = value.get("version_callback", None)
-    version_file = value.get("version_file", None)
-    count_commits_from_version_file = value.get("count_commits_from_version_file", False)
-    branch_formatter = value.get("branch_formatter", None)
-    sort_by = value.get("sort_by", None)
-
-    version = version_from_git(
-        template=template,
-        dev_template=dev_template,
-        dirty_template=dirty_template,
-        starting_version=starting_version,
-        version_callback=version_callback,
-        version_file=version_file,
-        count_commits_from_version_file=count_commits_from_version_file,
-        branch_formatter=branch_formatter,
-        sort_by=sort_by,
-    )
+    config = load_config_from_dict(value)
+
+    version = version_from_git(**config)
+
     dist.metadata.version = version
 
+    infer_version(dist)
```

Comment:
`dist.metadata.version` has already been set from `setup.py` config file, so it looks like `infer_version` should not be called

Reply:
I added this line to basically introduce `pyproject.toml` everywhere. If a user uses a `setup.py` and a `pyproject.toml` at the same time, I wanted to make sure the settings provided by `pyproject.toml` would be used by setuptools-git-versioning as well. The way it's current written means, that the settings in `pyproject.toml` have a higher priority than `setup.py`.

I chose this order because as far as I understood the python documentation of setuptools people should switch to `pyproject.toml` files. In turn this means if both files are present and settings are in `setup.py` the current implementation would ignore those and use default settings instead.

What behavior do you want? Which file should have a higher priority?

Reply:
That's ok. I just wasn't sure about calling `infer_version` after successfully setting up the version, but now I see that in case of missing `pyproject`.toml` it does not affect anything